ansible_collections/arista/avd/plugins/action/eos_validate_state_runner.py

```diff
@@ -0,0 +1,137 @@
+
+__metaclass__ = type
+
+import logging
+
+from ansible.errors import AnsibleActionFail
+from ansible.module_utils.common.arg_spec import ArgumentSpecValidator, ValidationResult
+from ansible.module_utils.errors import UnsupportedError
+from ansible.plugins.action import ActionBase, display
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils import AnsibleEOSDevice, get_anta_results
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import PythonToAnsibleContextFilter, PythonToAnsibleHandler
+
+LOGGER = logging.getLogger("ansible_collections.arista.avd")
+# ANTA 0.8.0 currently add some RichHandler to the root logger so need to disable propagation
+LOGGER.propagate = False
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        self._supports_check_mode = True
+
+        if task_vars is None:
+            task_vars = {}
+
+        validation_result, new_module_args = self._validate_argument_spec()
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        hostname = task_vars["inventory_hostname"]
+        ansible_connection = self._connection
+        ansible_check_mode = task_vars.get("ansible_check_mode", False)
+
+        # Handle logging
+        setup_module_logging(hostname, result)
+
+        # Get task arguments
+        logging_level = new_module_args["logging_level"]
+        save_catalog = new_module_args["save_catalog"]
+        skipped_tests = new_module_args.get("skipped_tests", {})
+        if save_catalog is True:
+            eos_validate_state_dir = new_module_args["device_catalog_output_dir"]
+            save_catalog_name = f"{eos_validate_state_dir}/{hostname}-catalog.yml"
+        else:
+            save_catalog_name = None
+
+        ansible_tags = {
+            "ansible_run_tags": task_vars.get("ansible_run_tags", ()),
+            "ansible_skip_tags": task_vars.get("ansible_skip_tags", ()),
+        }
+
+        # This is not all the hostvars, but just the Ansible Hostvars Manager object where we can retrieve hostvars for each host on-demand.
+        hostvars = task_vars["hostvars"]
+
+        try:
+            anta_device = AnsibleEOSDevice(name=hostname, connection=ansible_connection, check_mode=ansible_check_mode)
+            anta_results = get_anta_results(
+                anta_device=anta_device,
+                hostvars=hostvars,
+                logging_level=logging_level,
+                skipped_tests=skipped_tests,
+                ansible_tags=ansible_tags,
+                save_catalog_name=save_catalog_name,
+                # This convert Ansible Check Mode to dry_run
+                dry_run=ansible_check_mode,
+            )
+        except Exception as error:
+            raise AnsibleActionFail(message=str(error)) from error
+
+        result["results"] = anta_results
+
+        return result
+
+    def _validate_argument_spec(self) -> [ValidationResult, dict]:
```

Comment:
Strictly speakting the return type hint here is wrong. Should be parantheseses.

Reply:
the correct one would be `tuple[ValidationResult, dict]` but this is gone :)